app/questionnaire/router.py

```diff
@@ -161,16 +174,24 @@ def get_first_incomplete_location_for_section(
         self, routing_path, section_id, list_item_id=None
     ):
         section_key = (section_id, list_item_id)
+        first_incomplete_block_id = routing_path[0]
         if section_key in self._progress_store:
-            for location in routing_path:
-                if not self._is_location_complete(location):
-                    return location
-
-        return routing_path[0]
+            section_id = routing_path.section_id
+            list_item_id = routing_path.list_item_id
+            for block_id in routing_path:
+                if not self._is_location_complete(block_id, section_id, list_item_id):
+                    first_incomplete_block_id = block_id
+                    break
+
+        return Location(
+            block_id=first_incomplete_block_id,
+            section_id=routing_path.section_id,
+            list_item_id=routing_path.list_item_id,
+            list_name=routing_path.list_name,
+        )
```

Comment:
The `section_id` and `list_item_id` passed to this method are redundant if they are now properties of the `RouthingPath`.